lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
+    collection,
+    id,
+    content,
+    { refresh=null, userId=null, retryOnConflict=-1 })
+  {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      id,
+      refresh,
+      retryOnConflict
+    };
 
-    esRequest.id = request.input.resource._id;
-    esRequest.body = request.input.body;
+    // injecting retryOnConflict default configuration
+    if (esRequest.retryOnConflict === 0
+      && this.config.defaults.onUpdateConflictRetries > 0
+    ) {
+      esRequest.retryOnConflict = this.config.defaults.onUpdateConflictRetries;
+    }
 
     assertNoRouting(esRequest);
     assertWellFormedRefresh(esRequest);
 
-    return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)
+    return this.kuzzle.indexCache.exists(index, collection)
       .then(exists => {
         if (! exists) {
           this.throw(
             'index_or_collection_does_not_exist',
-            esRequest.index,
-            esRequest.type
-          );
-        }
-
-        // injecting retryOnConflict default configuration
-        if (!_.has(esRequest, 'retryOnConflict')
-          && this.config.defaults.onUpdateConflictRetries > 0
-        ) {
-          esRequest.retryOnConflict = this.config.defaults.onUpdateConflictRetries;
+            index,
+            collection);
         }
 
         // Add metadata
-        esRequest.body._kuzzle_info = {
-          active: true,
+        content._kuzzle_info = {
           updatedAt: Date.now(),
-          updater: this._getUserId(request)
+          updater: String(userId)
         };
 
-        esRequest.body = {doc: esRequest.body};
+        esRequest.body = { doc: content };
 
         return this.client.update(esRequest);
       })
-      .then(result => this.refreshIndexIfNeeded(esRequest, result))
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .then(({ body }) => {
+        const apiResponse = {
+          _id: body._id,
+          _version: body._version
+        };
+
+        return apiResponse;
+      })
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Replace a document to ElasticSearch
    *
-   * @param {Request} request
-   * @returns {Promise} resolve an object that contains _id
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {string} id - Document id
+   * @param {object} content - Document content
+   * @param {object} options - refresh (null), userId (null)
+   *
+   * @returns {Promise} { _id, _version, _source }
    */
-  replace(request) {
+  replace (index, collection, id, content,  { refresh=null, userId=null }) {
     const
-      esRequest = initESRequest(request, ['refresh']),
-      existQuery = {
-        index: esRequest.index,
-        type: esRequest.type,
-        id: request.input.resource._id
-      },
-      userId = this._getUserId(request);
+      esIndex = this._getESIndex(index, collection),
+      esRequest = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Group all shorthand properties at the end of this object declaration [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3499)